source/features/keyboard-navigation.tsx

```diff
@@ -14,8 +38,19 @@ function runShortcuts(event: KeyboardEvent): void {
 	if (['j', 'k'].includes(event.key)) {
 		event.preventDefault();
 
-		const items = select.all('.js-targetable-element:not([id^="pullrequestreview"]')
-			.filter(comment => !comment.querySelector('.minimized-comment:not(.d-none)'));
+		const items = select.all('.js-targetable-element')
+			.filter(element => {
+				if (element.id.startsWith('pullrequestreview-')) {
+					return pullRequestReviewHasComment(element);
+				}
+
+				if (element.classList.contains('js-minimizable-comment-group')) {
+					return !isCommentGroupMinimized(element);
+				}
+
+				return true;
+			});
```

Comment:
Maybe a longer selector is still faster than looping via JS
```suggestion
		const items = select.all([
			// Excludes reviews without a main comment
			'.js-targetable-element[id^="pullrequestreview"]:not(.js-comment)',
			
			// All targetable elements, except reviews and unnecessary events
			'.js-targetable-element:not([id^="pullrequestreview"]):not([id^="event-"])'
		]).filter(isCommentGroupMinimized);
```

Reply:
But maybe at this point we should _select_ exactly what we want, rather than select everything and then _filter_ it. Example

```suggestion
		const items = select.all([
			// Reviews with a main comment
			'[id^="pullrequestreview"][id$="-body-html"]',
			
			// Files in diffs
			'[id^="#diff-"]',
			
			// Comments (to be `.filter()`ed)
			'.js-minimizable-comment-group',
			
			// PR comments?
			'etc',
		])
		.filter(comment => !comment.querySelector('.minimized-comment:not(.d-none)'));
```

This is easier, even if the filter doesn't really apply to most of them

Reply:
I like that approach it still wouldn't solve *losing* the current target if it isn't in `items`.
But I noticed there are a lot of possible targets that we don't select anyways, so it might not be the best idea anyways.

The only thing I changed was
1. `#pullrequestreview-<id>-body-html` also have `.js-minimizable-comment-group` so I combined that.
2.  `[id^="diff-"]` was to general and also selected [lines in diffs](https://github.com/sindresorhus/refined-github/pull/3242/files#diff-0730bb7c2e8f9ea2438b52e419dd86c9R179) so I added `.js-targetable-element`.

I couldn't find any `.js-minimizable-comment-group:not(.js-targetable-element)` so I kept the simpler selector.